bin/dswx_landcover_mask.py

```diff
@@ -78,24 +88,126 @@ def _get_parser():
     return parser
 
 
+def point2epsg(lon, lat):
+    if lon >= 180.0:
+        lon = lon - 360.0
+    if lat >= 60.0:
+        return 3413
+    elif lat <= -60.0:
+        return 3031
+    elif lat > 0:
+        return 32601 + int(np.round((lon + 177) / 6.0))
+    elif lat < 0:
+        return 32701 + int(np.round((lon + 177) / 6.0))
+    raise ValueError(
+        'Could not determine projection for {0},{1}'.format(lat, lon))
+
 def main():
     parser = _get_parser()
 
     args = parser.parse_args()
 
     create_logger(args.log_file)
 
-    logger.info(f'Input file: {args.input_file}')
-    if not os.path.isfile(args.input_file):
-        logger.error(f'ERROR file not found: {args.input_file}')
+    if args.bbox is None and args.reference_file is None:
+        logger.error('ERROR please, provide either a '
+                     'reference file or a bounding box')
         return
-    layer_gdal_dataset = gdal.Open(args.input_file, gdal.GA_ReadOnly)
-    if layer_gdal_dataset is None:
-        logger.error(f'ERROR invalid input HLS file: {args.input_file}')
-    geotransform = layer_gdal_dataset.GetGeoTransform()
-    projection = layer_gdal_dataset.GetProjection()
-    length = layer_gdal_dataset.RasterYSize
-    width = layer_gdal_dataset.RasterXSize
+
+    if args.bbox is not None:
+        flag_is_geographic = True
+        lat_min, lat_max, lon_min, lon_max = args.bbox
+        # make sure lat_max > lat_min (since dy < 0)
+        if lat_max < lat_min:
+            lat_min, lat_max = lat_max, lat_min
+
+    geographic_srs = None
+    if args.reference_file:
+        print(f'Reference file: {args.reference_file}')
+
+        if not os.path.isfile(args.reference_file):
+            logger.error(f'ERROR file not found: {args.reference_file}')
+            return
+        layer_gdal_dataset = gdal.Open(args.reference_file, gdal.GA_ReadOnly)
+        if layer_gdal_dataset is None:
+            logger.error(f'ERROR invalid file: {args.reference_file}')
```

Comment:
If layer_gdal_dataset is None, then that error is logged to file, but the rest of the code is allowed to proceed. (But will quickly error-out, such as in line 136). Is there a missing `return` statement, similar to line 130? If not, what should the desired behavior be?